go/mysql/flavor_mysql.go

```diff
@@ -562,6 +562,8 @@ func (mysqlFlavor) setReplicationSourceCommand(params *ConnParams, host string,
 	}
 	if params.SslEnabled() {
 		args = append(args, "SOURCE_SSL = 1")
+	} else {
+		args = append(args, "GET_SOURCE_PUBLIC_KEY = 1")
```

Comment:
See https://dev.mysql.com/doc/refman/8.4/en/change-replication-source-to.html#crs-opt-get_source_public_key for what this flag does. It's only needed when SSL is not used for replication.